ScarletEngine/Sources/Gameplay/Public/World.h

```diff
@@ -45,6 +45,11 @@ namespace ScarletEngine
 			return *static_cast<System<SystemSig...>*>(Systems.emplace_back(GlobalAllocator<System<SystemSig...>>::New(&Reg, Name)).get());
 		}
 
+		void AddSystem(ISystem* System)
+		{
+			Systems.emplace_back(System);
```

Comment:
Ideally actually we should not be handling raw pointers here since this represents a transfer of ownership. I would prefer if we used `UniquePtr` as the parameter here and the code which constructs it would simply call `MakeUnique<SystemType>(params...)`.

Reply:
What's a UniquePtr?